site/content/3.12/data-science/graphrag/services/natural-language-to-aql.md

```diff
@@ -1,170 +1,297 @@
-API_PROVIDER=<provider>                # "openai" or "triton"
+# LLM Provider Configuration (chat model only)
+CHAT_API_PROVIDER=<openai>
 
-# If using OpenAI
-OPENAI_API_KEY=<your_api_key>
-OPENAI_MODEL=<model_name>              # Optional, defaults to GPT-4
-OPENAI_TEMPERATURE=<temperature>       # Optional
-OPENAI_MAX_RETRIES=<retries>           # Optional
+# If chat provider is OpenAI
+CHAT_API_KEY=<your_chat_api_key>
+CHAT_MODEL=<model_name>                # Optional, defaults to gpt-4o-mini (e.g., gpt-4o)
+CHAT_API_URL=<base_url>                # Optional, OpenAI‑compatible endpoint (e.g., https://openrouter.ai/api/v1)
 
-# If using Triton
-TRITON_URL=<triton_server_url>
-TRITON_MODEL=<model_name>
-TRITON_TIMEOUT=<timeout_seconds>       # Optional
 ```
 
-### Starting the Service
+### Provider-Specific Parameters
+
+#### OpenAI Provider (chat)
 
-To start the service, use GenAI service endpoint `CreateGraphRag`. Please refer to the documentation of GenAI service for more information on how to use it.
+- `chat_api_key`: API key for OpenAI chat
+- `chat_model`: Chat model (optional; defaults to `gpt-4o-mini`, e.g., "gpt-4o")
+- `chat_api_url` (optional): Override base URL for OpenAI‑compatible endpoints. This enables using providers like OpenRouter or self-hosted OpenAI-compatible models.
+- `openai_temperature` (optional): Controls randomness (0.0 to 2.0).
+- `openai_max_retries` (optional): Maximum number of retry attempts.
 
-### Required Parameters
+#### OpenRouter (via OpenAI‑compatible endpoint)
 
-These parameters must be provided in the install request sent to GenAI service.
+OpenRouter is supported by setting the OpenAI‑compatible base URL and using your OpenRouter API key. This allows access to many upstream LLM providers through a single API.
 
-- `username`: Database username for authentication
-- `db_name`: Name of the ArangoDB database
-- `api_provider`: LLM provider selection (`openai`, `triton`)
+```bash
+# Choose the OpenAI-compatible provider
+CHAT_API_PROVIDER=openai
 
-### Provider-Specific Required Parameters
+# Use your OpenRouter API key
+CHAT_API_KEY=<your_openrouter_api_key>
 
-#### OpenAI Provider
+# Point to OpenRouter's endpoint
+CHAT_API_URL=https://openrouter.ai/api/v1
 
-- `openai_api_key`: API key for OpenAI authentication
-- `openai_model`: Model name (defaults to "gpt-3.5-turbo" if not specified)
+# Select any model ID available on OpenRouter
+# (see OpenRouter's model catalog for valid IDs)
+CHAT_MODEL=<openrouter_model_id>
+```
 
-#### Triton Provider
+### Start the service
 
-- `triton_url`: URL of the Triton inference server
-- `triton_model`: Model name to use with Triton
+Create the service instance with your configuration:
 
-## API Reference
+```bash
+curl --request POST \
+  --url https://<ExternalEndpoint>:8529/ai/v1/graphrag \
+  --header 'Authorization: Bearer <your-bearer-token>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "env": {
+      "username": "<your-username>",
```

Comment:
Note: `username` is currently still required for startup, but we are working on removing this requirement (following the same approach in the importer and retriever).